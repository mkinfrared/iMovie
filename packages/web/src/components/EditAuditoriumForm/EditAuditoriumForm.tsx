import React from "react";

import { EditAuditoriumFormProps } from "./EditAuditoriumForm.type";

const EditAuditoriumForm = (_: EditAuditoriumFormProps) => {
  return <div>EditAuditoriumForm</div>;
};

export default React.memo(EditAuditoriumForm);
