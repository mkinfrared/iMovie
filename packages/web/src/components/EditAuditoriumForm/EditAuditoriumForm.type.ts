export interface EditAuditoriumFormProps {
  id: number;
}
