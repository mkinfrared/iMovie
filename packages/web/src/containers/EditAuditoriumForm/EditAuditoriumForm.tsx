import React from "react";
import { useSelector } from "react-redux";

import EditAuditoriumFormFC from "components/EditAuditoriumForm";
import { EditAuditoriumFormProps } from "components/EditAuditoriumForm/EditAuditoriumForm.type";
import { selectCinemaAuditorium } from "store/reducers/cinemas/selectors";

const EditAuditoriumForm = ({
  auditoriumId,
  cinemaId,
  ...rest
}: Omit<EditAuditoriumFormProps, "auditorium">) => {
  const auditorium = useSelector(
    selectCinemaAuditorium(cinemaId, auditoriumId)
  );

  if (!auditorium) {
    throw new Error("Auditorium was not found");
  }

  return (
    <EditAuditoriumFormFC
      {...rest}
      auditoriumId={auditoriumId}
      cinemaId={cinemaId}
      auditorium={auditorium}
    />
  );
};

export default React.memo(EditAuditoriumForm);
