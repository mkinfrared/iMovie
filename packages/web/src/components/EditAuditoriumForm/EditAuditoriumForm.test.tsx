import { render } from "@testing-library/react";
import React, { ReactElement } from "react";

import EditAuditoriumForm from "./EditAuditoriumForm";

describe("<EditAuditoriumForm />", () => {
  let Component: ReactElement;

  beforeEach(() => {
    Component = <EditAuditoriumForm />;
  });

  it("should be defined", () => {
    const { container } = render(Component);

    expect(container).toBeDefined();
  });

  it("should match a snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
