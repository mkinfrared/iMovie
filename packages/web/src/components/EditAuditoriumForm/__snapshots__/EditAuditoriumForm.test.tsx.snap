// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditAuditoriumForm /> should match a snapshot 1`] = `
<div>
  <div>
    EditAuditoriumForm
  </div>
</div>
`;
